src/UKHO.ERPFacade.API/Controllers/WebhookController.cs

```diff
@@ -157,6 +159,23 @@ public virtual async Task<IActionResult> RecordOfSalePublishedEventReceived([Fro
             await _azureBlobEventWriter.UploadEvent(recordOfSaleEventJson.ToString(), RecordOfSaleContainerName, correlationId + '/' + RecordOfSaleEventFileName);
             _logger.LogInformation(EventIds.UploadedRecordOfSalePublishedEventInAzureBlob.ToEventId(), "Record of sale published event is uploaded in blob storage successfully.");
 
+            string recordOfSalesXmlTemplatePath = Path.Combine(Environment.CurrentDirectory, RosXmlFilePath);
+
+            XmlDocument rosPayload = new();
+            rosPayload.Load(recordOfSalesXmlTemplatePath);
+
+            HttpResponseMessage response = await _sapClient.PostEventData(rosPayload, _sapConfig.Value.SapServiceOperationForRecordOfSale, _sapConfig.Value.SapUsernameForRecordOfSale, _sapConfig.Value.SapPasswordForRecordOfSale);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                _logger.LogError(EventIds.ErrorOccuredInSap.ToEventId(), "An error occured while processing your request in SAP. | {StatusCode}", response.StatusCode);
```

Comment:
Please use new Event id in case of error occurred while processing Record of Sale endpoint.

Reply:
New event Id "ErrorOccurredInSapForRecordOfSalePublishedEvent" added for record of sale published event endpoint.